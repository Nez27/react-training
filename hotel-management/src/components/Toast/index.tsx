import { DefaultToastOptions, ToastOptions, ToastPosition, Toaster } from 'react-hot-toast';
import { CSSProperties } from 'styled-components';

interface IToast {
  position: ToastPosition;
  gutter: number;
  containerStyle: CSSProperties;
  toastOptions: DefaultToastOptions;
  success: ToastOptions;
  error: ToastOptions;
  style: CSSProperties | undefined;
}

const Toast = ({position, gutter, containerStyle, success, error, style}: IToast) => {
  return (
    <Toaster
      position={position}
      gutter={gutter}
      containerStyle={containerStyle}
      toastOptions={{
        success,
        error,
        style
      }}
    />
  );
};

Toast.defaultProps = {
  position: "top-center",
  gutter: 12,
  containerStyle: { margin: '8px', zIndex: 1 },
  success: {
    duration: 3000,
  },
  error:{
    duration: 5000,
  },
  style: {
    fontSize: '16px',
    maxWidth: '500px',
    padding: '16px 24px',
  }
}

export default Toast;
