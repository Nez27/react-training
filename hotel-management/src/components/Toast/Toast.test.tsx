import renderer from 'react-test-renderer';

// Components
import Toast from '.';

describe('Toast', () => {
  const wrapper = renderer.create(<Toast />);

  test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
