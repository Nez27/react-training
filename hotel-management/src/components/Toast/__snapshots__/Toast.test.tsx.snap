// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Should render correctly 1`] = `
<div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    {
      "bottom": 16,
      "left": 16,
      "margin": "8px",
      "pointerEvents": "none",
      "position": "fixed",
      "right": 16,
      "top": 16,
      "zIndex": 2000,
    }
  }
/>
`;
